backend/src/question/question.controller.ts

```diff
@@ -0,0 +1,48 @@
+import { Controller, Get, Param, Query } from "@nestjs/common";
+
+import { QuestionService } from "./question.service";
+import { PublicRoute } from "../utils/decorator";
+
+@Controller("question")
+export class QuestionController {
+  constructor(private readonly questionService: QuestionService) {}
+
+  @PublicRoute()
+  @Get("/content/:titleSlug")
+  async getContentFromSlug(@Param("titleSlug") titleSlug: string) {
+    const questionContent = await this.questionService.getContentFromSlug(
+      titleSlug
+    );
+    return questionContent;
+  }
+
+  @PublicRoute()
+  @Get("/summaries/all")
+  async getSummeries() {
+    const questionSummaries = await this.questionService.getSummaries();
+    return questionSummaries;
+  }
+
+  //TODO: Ensure that the routes below work
+  //TODO: Ensure that string[] is being passed, currently if string[].length == 1,
+  //TODO:  NodeJS takes it as a char[].
+
+  @PublicRoute()
+  @Get("/summaries")
+  async getSummariesFromSlug(@Query() titleSlugs: { titleSlug: string[] }) {
+    const filteredSummaries = await this.questionService.getSummaryFromSlug(
+      titleSlugs
+    );
+
+    return filteredSummaries;
+  }
+
+  @PublicRoute()
+  @Get("/summaries")
+  async getSummariesFromTopicTags(@Query() topics: { topic: string[] }) {
```

Comment:
should be `Query() topics: string[] | string | undefined`

Reply:
I updated this to use `params` since (I think) it's easier to handle, but let me know what's the preferred way/best practice.

Reply:
should use query since it allows for an array of topic tags